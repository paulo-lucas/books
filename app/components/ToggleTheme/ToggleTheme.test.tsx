import 'react-native';
import React from 'react';
import { fireEvent, render, screen } from '../../../tests/test-utils';
import { ToggleTheme } from './';

const renderToggleTheme = () => render(<ToggleTheme />);

describe('toggle theme button', () => {
  it('should render correctly', () => {
    renderToggleTheme();
    expect(screen.toJSON()).toMatchSnapshot('toggle theme button');
  });

  it('should render correctly when toggled', () => {
    renderToggleTheme();

    fireEvent(screen.getByTestId('toggle-theme-button'), 'onPress');

    expect(screen.toJSON()).toMatchSnapshot('toggle theme button dark mode');
  });
});
