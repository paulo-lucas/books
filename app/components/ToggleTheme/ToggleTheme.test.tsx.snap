// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggle theme button should render correctly when toggled: toggle theme button dark mode 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Object {
      "backgroundColor": undefined,
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "overflow": "hidden",
      }
    }
    testID="RNE__ICON__CONTAINER"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      replaceTheme={[Function]}
      testID="toggle-theme-button"
      updateTheme={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
        testID="RNE__ICON"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#CDCDCD",
                "fontSize": 24,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="RNE__ICON__Component"
        >
          
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`toggle theme button should render correctly: toggle theme button 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Object {
      "backgroundColor": undefined,
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "overflow": "hidden",
      }
    }
    testID="RNE__ICON__CONTAINER"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      replaceTheme={[Function]}
      testID="toggle-theme-button"
      updateTheme={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
        testID="RNE__ICON"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#242424",
                "fontSize": 24,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="RNE__ICON__Component"
        >
          
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
